lib/inject.rb

```diff
@@ -0,0 +1,8 @@
+class Array
+
+	def sebastien_injection(a= nil, b=nil)
+		injectme = a ? b : self.shift
+		self.map {|x| injectme = yield(injectme , x)}
+		return injectme	
```

Comment:
Ruby doesn't need an explicit `return` - the last line evaluated will be returned, so in this case you could just type `injectme`